twarep/electrifly
# Storage script

## Checklist

## What I Did
Created a file called storage.py. It contains a bunch of methods that relate to pushing data into our database. For when the database connection string changes, the only thing that will have to change is the DATABASE_URL in the .env file, saving us a lot of work.

Modified transformation.py and scraper.py where appropriate.

## How I Did it?
Used SQLAlchemy, since pandas doesn't let you push data frames into a database unless the connection engine is via SQLAlchemy.

## How to Test?
You will need to setup the repository in linux. The instructions to do that is the same as in the readme, except you have to create the virtual environment like this: `python3.9 -m venv venv` to ensure that the python version is 3.9 (the minimum needed to run our app)
1. Delete the entire database using DROP SCHEMA public CASCADE;
2. Recreate the database using CREATE SCHEMA public;
3. Run scraper.py (I removed the pipeline directory, so just run it within the electrifly directory)
4. Check the database in DBeaver. It should be fully populated, looking like this:
<img width="1203" alt="Screen Shot 2023-07-20 at 7 39 43 PM" src="https://github.com/twarep/electrifly/assets/84249628/9905fd30-98b8-4bb1-8d4f-0a4088864ef9">
If you need help connecting to our database using DBeaver, see the Notion ticket.
